### PR TITLE
[PLT-5864] Move teammate name display setting to the System Console

### PR DESCRIPTION
**Pull request by saturninoabril**
#### Summary
Move teammate name display setting to the System Console

#### Ticket Link
Jira ticket: [PLT-5864](https://mattermost.atlassian.net/browse/PLT-5864)

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Added or updated unit tests (required for all new features)
- [x] Has UI changes
- [x] Includes text changes and localization file ([.../i18n/en.json](https://github.com/mattermost/platform/blob/master/i18n/en.json) and [.../webapp/i18n/en.json](https://github.com/mattermost/platform/tree/master/webapp/i18n/en.json)) updates

### CODE REVIEW

File: app/diagnostics.go [hunk 1]
```diff
@@ -208,6 +208,7 @@ func trackConfig() {
 		"max_notifications_per_channel":       *utils.Cfg.TeamSettings.MaxNotificationsPerChannel,
 		"max_users_per_team":                  utils.Cfg.TeamSettings.MaxUsersPerTeam,
 		"max_channels_per_team":               *utils.Cfg.TeamSettings.MaxChannelsPerTeam,
+		"teammate_display":                    *utils.Cfg.TeamSettings.TeammateDisplay,
```
Comment (jwilander) [inline app/diagnostics.go]: Can we use `TeammateNameDisplay` for the config setting?

File: webapp/components/admin_console/users_and_teams_settings.jsx [hunk 2]
```diff
@@ -188,6 +190,28 @@ export default class UsersAndTeamsSettings extends AdminSettings {
...
+                    label={
+                        <FormattedMessage
+                            id='admin.team.teammateDisplay'
+                            defaultMessage='Teammate Name Display:'
+                        />
+                    }
+                    helpText={
+                        <FormattedHTMLMessage
+                            id='admin.team.teammateDisplayDesc'
+                            defaultMessage="Set how to display user's names in posts and the Direct Messages list."
```
Comment (jwilander) [inline webapp/components/admin_console/users_and_teams_settings.jsx]: Grammar mistake, I think `user's` should be `users'`, @jasonblais (you opened the ticket, sorry if you're not the PM on this)

Comment (jasonblais) [inline webapp/components/admin_console/users_and_teams_settings.jsx]: Thanks @jwilander , sounds good

**Comment by jwilander** [comment]
Looks good but you have a gofmt error:
```
Running GOFMT
diff model/config.go gofmt/model/config.go --- /tmp/gofmt103162942	2017-06-20 23:50:55.704602999 +0000 +++ /tmp/gofmt265510533	2017-06-20 23:50:55.704602999 +0000 @@ -331,7 +331,7 @@  	UserStatusAwayTimeout               *int64  	MaxChannelsPerTeam                  *int64  	MaxNotificationsPerChannel          *int64 -	TeammateNameDisplay                     *string +	TeammateNameDisplay                 *string  }    type LdapSettings struct {
gofmt failure
```

**Comment by grundleborg** [comment]
The UX aspects of this change seem to have been reopened for discussion, so I won't merge it just yet.

See https://pre-release.mattermost.com/core/pl/eeuatiaodb865dtkmc7qnjtmua

**Comment by saturninoabril** [comment]
Done rebase. There's a proposal to keep with (original) current plan which is moving to System console. That is covered by this PR.

**Pull request merged**